modules/templates/helper.go

```diff
@@ -339,33 +339,37 @@ func RenderCommitMessageLink(msg, urlPrefix, urlDefault string, metas map[string
 // the provided default url, handling for special links without email to links.
 func RenderCommitMessageLinkSubject(msg, urlPrefix, urlDefault string, metas map[string]string) template.HTML {
 	cleanMsg := template.HTMLEscapeString(msg)
+
+	msgLines := strings.Split(strings.TrimSpace(string(cleanMsg)), "\n")
```

Comment:
Seems a bit wasteful to escape the whole string, compute the complete split and trim the whole lot if we're just going to use the first line.

How about (although this probably counts as premature optimisation):

```go
msgLine := strings.TrimLeftFunc(msg, unicode.IsSpace)
if strings.IndexOf('\n') > 0 {
        msgLine = msgLine[:strings.IndexOf('\n') - 1]
}
msgLine := string.TrimRightFunc(msgLine, unicode.IsSpace)
if len(msgLine) == 0 {
        return template.HTML("")
}
msgBytes := template.HTMLEscapeString(msgLine)

// we can safely assume that it will not return any error, since there
// shouldn't be any special HTML.
renderedMessage, err := markup.RenderCommitMessageSubject(msgBytes, urlPrefix, urlDefault, metas)
```

Reply:
ok, did code optimizations in both functions ;)